src/render/src/encoder.rs

```diff
@@ -167,14 +167,15 @@ impl<R: Resources, C: command::Buffer<R>> Encoder<R, C> {
     /// internal ´CommandBuffer´ will not be sent to the GPU, and as a result they will not be
     /// processed. Calling flush too often however will result in a performance hit. It is
     /// generally recommended to call flush once per frame, when all draw calls have been made. 
-    pub fn flush<D>(&mut self, device: &mut D) where
-        D: Device<Resources=R, CommandBuffer=C>
+    pub fn flush<D>(&mut self, device: &mut D) -> SubmissionResult<()>
```

Comment:
I'd like the `flush()` to just be a helper function, thus not returning anything.
One should be able to do `device.submit()` followed by `encoder.reset()` if they want more control and/or error processing, so let's have `flush()` be convenient.